src/resource/factory/DisplayListFactory.cpp

```diff
@@ -145,24 +145,29 @@ std::shared_ptr<Ship::IResource> ResourceFactoryBinaryDisplayListV0::ReadResourc
         reader->ReadInt8();
     }
 
+    size_t idx = 0;
     while (true) {
         Gfx command;
         command.words.w0 = reader->ReadUInt32();
         command.words.w1 = reader->ReadUInt32();
 
-        displayList->Instructions.push_back(command);
-
-        uint8_t opcode = (uint8_t)(command.words.w0 >> 24);
+        int8_t opcode = (int8_t)(command.words.w0 >> 24);
```

Comment:
this change from `uint8_t` to `int8_t` hiding in this PR feels a little scary to me. if using `uint8_t` here was a mistake i'd feel better having a separate PR fixing that so we can discuss that change, then this PR can focus solely on *adding* a feature.

Reply:
I think that was a mistake i did because i copy/paste some things